packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -82,6 +82,18 @@ export function isExplicitImportRequired(url: string): boolean {
   return !isJSRequest(cleanUrl(url)) && !isCSSRequest(url)
 }
 
+function isFileExists(filepath: string): boolean {
+  const dir = path.dirname(filepath)
+  if (dir === path.dirname(dir)) {
+    return true
+  }
+  const filenames = fs.readdirSync(dir)
+  if (!filenames.includes(path.basename(filepath))) {
+    return false
+  }
+  return isFileExists(dir)
+}
```

Comment:
This looks quite costly, would you expand more on what you are trying to check here?

Reply:
Check like this

```
- users
-- vite-project // vite dev here, so the `root` is /users/vite-project
--- node_modules // exclude it
--- public/** // check
--- src/** // check
--- vite.config.ts // check

-- other-project // Exclude it, only check the files in the root directory
--- ***
```

Reply:
There isn't a way to obtain the actual case of a path in Node.js and it seems it won't be implemented in Node.js/libuv (https://github.com/nodejs/node/issues/24942).

So to avoid this recursive `fs.readdir` we'll need to create a native package in c/rust. (or find one?)